### PR TITLE
Issue #1285: Include an editor role in the standard profile

### LINKED ISSUE #1285: Include an "Editor" role in the standard install profile
Looking at the default roles, you'd think there are no content creators in the Bd world.

roles

Some roles to possibly add. Not sure what is the best name
1. contributor, writer, author?
   - give this role the ability to create and edit their own content.
   - No admin stuff.
   - Can unpublish their own content but not delete.
2. editor
   - edit any content regardless of authorship.
   - has the power to unpublish and delete any content.
   - power to disable/freeze any contributor account (role 1 above) from submitting new content. Basically, if contributor is being bad or incompetent, the content will remain but since account closed, they won't be able to work on the site anymore.
3. guest poster
   - can only create unpublished content

Dunno about the last 2 roles but throwing them out there

---
~PR by @stpaultim: https://github.com/backdrop/backdrop/pull/2830~
Updated PR by @quicksketch: https://github.com/backdrop/backdrop/pull/2837

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/1285

Replaces https://github.com/backdrop/backdrop/pull/2830

### CODE REVIEW

File: core/profiles/standard/standard.install [hunk 1]
```diff
@@ -346,14 +346,68 @@ function standard_install() {
...
+    'access comments',
+    'post comments',
+    'skip comment approval',
+    'upload editor images',
+    $filtered_html_permission,
+  );
+  user_role_grant_permissions(BACKDROP_ANONYMOUS_ROLE, $anonymous_permissions);
+  user_role_grant_permissions(BACKDROP_AUTHENTICATED_ROLE, $authenticated_permissions);
+
+  // Create a default editor role for users who need to be able to edit content
```
Comment (klonos) [inline core/profiles/standard/standard.install]:
```suggestion
  // Create a default editor role, for users who need to be able to edit
```

File: core/profiles/standard/standard.install [hunk 2]
```diff
@@ -346,14 +346,68 @@ function standard_install() {
...
+  // but do not need to access advanced admin features.
```
Comment (klonos) [inline core/profiles/standard/standard.install]:
```suggestion
  // content, but do not need to access advanced admin features.
```

File: core/profiles/standard/tests/standard.test [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    $edit['mail']   = $edit['name'] . '@example.com';
+    $edit['pass']   = user_password();
+    $edit['status'] = 1;
+    $edit['roles'] = array('editor');
+
+    $editor_account = entity_create('user', $edit);
+    $editor_account->save();
+    $editor_account->pass_raw = $edit['pass'];
+
+    // Just check that thr role exists and that the user can see the dashboard.
```
Comment (klonos) [inline core/profiles/standard/tests/standard.test]: Typo:
```suggestion
    // Just check that the role exists and that the user can see the dashboard.
```

File: core/profiles/standard/tests/standard.test [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  /**
+   * Tests existence of test case located in an installation profile module.
+   */
+  function testRolesExist() {
+    $user_1 = user_load(1);
+    $this->assertTrue(user_has_role('administrator', $user_1), 'User ID 1 is an administrator');
+
+    // Create a user assigned to the editor role.
+    $edit = array();
+    $edit['name']   = $this->randomName();
```
Comment (klonos) [inline core/profiles/standard/tests/standard.test]: I know it looks pretty, but our coding standards (and my OCDs) say no 😄 

```suggestion
    $edit['name'] = $this->randomName();
```

File: core/profiles/standard/tests/standard.test [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    $edit['mail']   = $edit['name'] . '@example.com';
```
Comment (klonos) [inline core/profiles/standard/tests/standard.test]:
```suggestion
    $edit['mail'] = $edit['name'] . '@example.com';
```

File: core/profiles/standard/tests/standard.test [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    $edit['pass']   = user_password();
```
Comment (klonos) [inline core/profiles/standard/tests/standard.test]:
```suggestion
    $edit['pass'] = user_password();
```

File: core/profiles/standard/tests/standard.test (hunk 4 above)
Comment (quicksketch) [reply core/profiles/standard/tests/standard.test]: Yeah, unintentional. Copy/paste from elsewhere. :P

**Comment by quicksketch** [comment]
Merged into 1.x for 1.14.0.

**Pull request merged**